app/finders/songs_finder.rb

```diff
@@ -1,6 +1,6 @@
 class SongsFinder
 
-  def self.search(keywords)
+  def self.search(keywords, key, tempo)
```

Comment:
So I think here, after the full text search, you can filter out the songs that don't have the right key or tempo, assuming they are set.